src/ucp/wireup/wireup.c

```diff
@@ -1220,14 +1220,34 @@ int ucp_wireup_is_reachable(ucp_ep_h ep, unsigned ep_init_flags,
                             ucp_rsc_index_t rsc_index,
                             const ucp_address_entry_t *ae)
 {
-    ucp_context_h context      = ep->worker->context;
-    ucp_worker_iface_t *wiface = ucp_worker_iface(ep->worker, rsc_index);
-
-    return (context->tl_rscs[rsc_index].tl_name_csum == ae->tl_name_csum) &&
-           (/* assume reachability is checked by CM, if EP selects lanes
-             * during CM phase */
-            (ep_init_flags & UCP_EP_INIT_CM_PHASE) ||
-            uct_iface_is_reachable(wiface->iface, ae->dev_addr, ae->iface_addr));
+    int result;
+    ucp_context_h context                        = ep->worker->context;
+    ucp_worker_iface_t *wiface                   = ucp_worker_iface(ep->worker, rsc_index);
+    const uct_iface_is_reachable_params_t params = {
+        .device_addr        = ae->dev_addr,
+        .iface_addr         = ae->iface_addr,
+        .info_string        = (char*)ucs_malloc(UCP_WIREUP_UNREACHABLE_INFO_STRING_DEFAULT_LENGTH,
+                                                "wireup diag"),
+        .info_string_length = UCP_WIREUP_UNREACHABLE_INFO_STRING_DEFAULT_LENGTH
+    };
+
+    if (params.info_string == NULL) {
+        ucs_error("failed to allocate memory for uct_iface_is_reachable_params_t param");
+        return UCS_ERR_NO_MEMORY;
+    }
+    
+    params.info_string[0] = '\0';
+    result = ((context->tl_rscs[rsc_index].tl_name_csum == ae->tl_name_csum) &&
+              (/* assume reachability is checked by CM, if EP selects lanes
+                * during CM phase */
+               (ep_init_flags & UCP_EP_INIT_CM_PHASE) ||
+               uct_iface_is_reachable_v2(wiface->iface, &params)));
+    if (!result && params.info_string != NULL && params.info_string[0] != '\0') {
```

Comment:
Why need to check the condition params.info_string[0] != '\0'?

Reply:
Avoid to outputting empty diagnostic log message